### ISSUE TITLE
Add `expectedCyclesToComplete` to `assertChanged` in RecipeTest

### ISSUE THREAD

**Issue by jkschneider** [openrewrite/rewrite#487]
To prove not only that recipes produce textually consistent results over any number of cycles, but that after making some set of changes, that it only takes some certain number of cycles to produce this.

I suspect that some recipes are changing things like `List` references over multiple cycles that don't affect the textual outcome of the recipe, but do cause further cycles to run unnecessarily.

I think the default for `expectedCyclesToComplete == cycles - 1`, unless the recipe specifically indicates that it wouldn't require another cycle even if it makes changes (see #486) in which case `expectedCyclesToComplete == 1` by default. 

In other words, the check validates that the recipe does _not_ make changes on the last cycle ran.
